packages/widget-v2/src/state/swapPage.ts

```diff
@@ -128,38 +125,6 @@ export const invertSwapAtom = atom(null, (get, set) => {
   }
 });
 
-export const routeAmountEffect = atomEffect((get, set) => {
-  const route = get(skipRouteAtom);
-  const direction = get(swapDirectionAtom);
-  const sourceAsset = get(sourceAssetAtom);
-  const destinationAsset = get(destinationAssetAtom);
-
-  if (!route.data || !sourceAsset || !destinationAsset) return;
-
-  const swapInAmount = convertTokenAmountToHumanReadableAmount(
-    route.data.amountOut,
-    destinationAsset.decimals
-  );
-  const swapOutAmount = convertTokenAmountToHumanReadableAmount(
-    route.data.amountIn,
-    sourceAsset.decimals
-  );
-  const swapInAmountChanged = swapInAmount !== destinationAsset.amount;
-  const swapOutAmountChanged = swapOutAmount !== sourceAsset.amount;
-
-  if (direction === "swap-in" && swapInAmountChanged) {
-    set(destinationAssetAtom, (old) => ({
-      ...old,
-      amount: limitDecimalsDisplayed(swapInAmount),
-    }));
-  } else if (direction === "swap-out" && swapOutAmountChanged) {
-    set(sourceAssetAtom, (old) => ({
-      ...old,
-      amount: limitDecimalsDisplayed(swapOutAmount),
-    }));
-  }
-});
-
```

Comment:
I believe this was the main culprit causing the stack size exceeded / infinite set state

Reply:
I see, nice catch